.travis.yml

```diff
@@ -1,16 +1,16 @@
 language: csharp
 mono: latest
-dotnet: 2.1
+dotnet: 5.0
 sudo: required
 env:
   global:
     - NUODB_HOME=/opt/nuodb
     - NUO_ADD_DOMAIN_PASS=yes
     - NUO_START_AGENT=yes
   matrix:
-    - NUO_VERSION=3.2.5.1
-    - NUO_VERSION=3.3.2.2
-    - NUO_VERSION=3.4.1.1
+    - NUO_VERSION=4.0.8.2
+    - NUO_VERSION=4.1.3.1
+    - NUO_VERSION=4.2.2.2
 
 before_install:
   - echo madvise | sudo tee /sys/kernel/mm/transparent_hugepage/enabled >/dev/null
```

Comment:
Frustratingly I can't comment on the parts of the code that have NOT been changed... what kind of crappy UI is that?

Anyway, I'm pretty sure you'll need to change the download URL in the wget command below.  Does that work for you if you try that wget?

See the example travis file in the nuodb-python GitHub repo for a URL that I believe works.

Reply:
Also, all the scripting needs to be rewritten to use nuoadmin because the startup/etc. here is all written for old admin which is obsolete in 4.1 and not even shipped anymore in 4.2+.  Again I recommend basing your travis file on the one for nuodb-python project which is a bit more up to date.

Reply:
Okay, sounds good. I will make those changes